mods/player_api/api.lua

```diff
@@ -60,37 +58,46 @@ function player_api.set_model(player, model_name)
 			eye_height = 1.625,
 		})
 	end
-	player_model[name] = model_name
+	player_data.model = model_name
 end
 
 function player_api.set_textures(player, textures)
-	local name = player:get_player_name()
-	local model = models[player_model[name]]
-	local model_textures = model and model.textures or nil
-	player_textures[name] = textures or model_textures
-	player:set_properties({textures = textures or model_textures})
+	local player_data = players[player:get_player_name()]
+	local model = models[player_data.model]
+	local new_textures = model and model.textures or textures
+	player_data.textures = new_textures
+	player:set_properties({textures = new_textures})
 end
 
 function player_api.set_animation(player, anim_name, speed)
-	local name = player:get_player_name()
-	if player_anim[name] == anim_name then
+	local player_data = players[player:get_player_name()]
+	local model = models[player_data.model]
+	if not (model and model.animations[anim_name]) then
 		return
 	end
-	local model = player_model[name] and models[player_model[name]]
-	if not (model and model.animations[anim_name]) then
+	speed = speed or model.animation_speed
+	if player_data.animation == anim_name and player_data.animation_speed == speed then
 		return
 	end
 	local anim = model.animations[anim_name]
-	player_anim[name] = anim_name
-	player:set_animation(anim, speed or model.animation_speed, animation_blend)
+	player_data.animation = anim_name
+	player_data.animation_speed = speed
+	player:set_animation(anim, speed, animation_blend)
+	player:set_properties({
```

Comment:
Ideally, this should only be executed if the collisionbox or eye_height changes (as in my PR). With the MTG player model, these only change when sitting or laying.

Reply:
Engine responsibility. See minetest/minetest#9934

Reply:
Hmm, ideally yes, but the engine does not do that yet, and with our slow dev may not do so for years.
I would like to avoid setting properties and sending packets every time a player starts or stops walking or mining, which is very often.
Please could you attempt this?
But i will not insist, someone else could add that later.

Reply:
I see three options:

1. Don't do it. Higher code quality, unnecessary packets in practice. Issue can be fixed better by the engine.
2. Do it, to avoid those packets, annotate as hack.
    1. Compare object properties, requires get_properties call and relatively cheap comparison, will override values changed by other mods
    2. Compare cached values, faster, neater, won't override values if changed by other mods (probably desirable)

We might want to settle on 2.2.

Reply:
I think it is best we wait for input from another core dev concerning what we do here.

Reply:
I suggest 2., whether i. or ii. doesn't really matter. Longterm the engine needs a proper `player_monoids`-esque solution to deal with multiple mods want to set something.

Reply:
FWIW. I agree with (2). Yes, it should be engine optimizing this, but until we have that, let's have mods be smart.
(See my NSSM change to avoid hammering the server with unnecessary node updates... Even though the server should avoid applying no-op changes).

Reply:
In case it is not obvious, i support 2 also.

Reply:
See https://github.com/minetest/minetest_game/pull/2745#issuecomment-706591210